File: source code/visitor_register.js
function validate()
	{
		var username = document.forms["form_register"]["user_name"].value;
		var f_name = document.forms["form_register"]["first_name"].value;
		var l_name = document.forms["form_register"]["last_name"].value;
		var pwd = document.forms["form_register"]["password"].value;
		var pwd2 = document.forms["form_register"]["password2"].value;
		var mail = document.forms["form_register"]["e_mail"].value;
		var phone = document.forms["form_register"]["phone_no"].value;
		var adr = document.forms["form_register"]["address"].value;
		if (username == "" || f_name == "" || l_name == "" || pwd == "" || pwd2 == "" || mail == "" || phone == "" || adr == "")
		{
			window.alert("All boxes should be filled!");
			return false;
		}
		else if (username.includes(" ") || f_name.includes(" ") || l_name.includes(" ") || pwd.includes(" ") ||
		pwd2.includes(" ") || mail.includes(" ") || phone.includes(" ") || adr.includes(" "))
		{
		    window.alert("Information must be entered without any space");
		    return false;
		}
		else if (pwd != pwd2)
		{
		    window.alert("Passwords don't match");
		    return false
		}
		return true;
	}

File: source code/visitor_register.py
#!C:\Users\sergu\AppData\Local\Programs\Python\Python310\python.exe
import sys


def print_header(title):
    print("Content-type: text/html")
    print("")
    print("""<html><head><title>{}</title>
    <script src = "visitor_register.js"></script>
    <link rel="stylesheet" type="text/css" href="visitor_register.css"></head><body>
    <ul>
  <li style="background-color:rgb(223, 223, 223)"><a style="color:red; font-family:Papyrus,Fantasy" href="visitor.py">YMS</a></li>
  <li style="float:right"><a href="visitor_login.py">LogIn</a></li>
  </ul>""".format(title))


def print_form():
    print("""
    <h2></h2>
    <div>
    <h2 style="color:black; text-align: center">Registration form</h2>
    <form name = "form_register" method="post" action= "user_register.py" onsubmit="return validate();">
    <label for="first_name">First name:</label><br>
    <input type="first_name" id="first_name" name="first_name">
    <label for="last_name">Surname:</label><br>
    <input type="last_name" id="last_name" name="last_name">
    <label for="user_name">User name:</label><br>
    <input type="text" id="user_name" name="user_name"><br>
    <label for="password">Password:</label><br>
    <input type="password" id="password" name="password">
    <label for="password2">Re-enter password:</label><br>
    <input type="password" id="password2" name="password2">
    <label for="e_mail">E-mail:</label><br>
    <input type="e_mail" id="e_mail" name="e_mail">
    <label for="phone_no">Phone number:</label><br>
    <input type="phone_no" id="phone_no" name="phone_no">
    <label for="address">Address:</label><br>
    <input type="address" id="address" name="address">
    <p><input type="submit" value="Register" class="button button1"></p>
    </form>
    </div>""")


def print_footer():
    print("</body></html>")


if __name__ == '__main__':
    print_header("Register")
    print_form()
    print_footer()
